src/sentry/search/events/builder/metrics.py

```diff
@@ -111,30 +112,32 @@ def _resolve_on_demand_spec(
             return None
 
     def _get_on_demand_metrics_query(self) -> Optional[MetricsQuery]:
-        if not self.is_performance or not self.is_alerts_query:
-            return None
-
         spec = self._on_demand_spec
 
+        # TimeseriesQueryBuilder specific parameters
+        if self.is_timeseries_query:
+            limit = None
+            alias = "count"
+        else:
+            limit = self.limit
+            alias = spec.mri
```

Comment:
Instead of defining a boolean value that isn't very idiomatic and a bit hacky, can we leverage some nice method overloading by defining a method `get_query_params` or something similar which in this class might have the following impl:

```
@dataclass(frozen=True)
class QueryParams:
	limit: Optional[int]
	alias: str
	

def get_query_params():
	return QueryParams(
		limit: self.limit,
		alias: self._on_demand_spec.mri
	)
```

What do you think?

Reply:
Then in the `TimeseriesMetricQueryBuilder` you can override it as:

```
def get_query_params():
	return QueryParams(
		limit: None,
		alias: "count"
	)
```

Reply:
Yep, I added another flag similar to https://github.com/getsentry/sentry/blob/ef71fd41b8a00dd3c12b6799db019ba6c4642559/src/sentry/search/events/builder/metrics.py#L52C5-L52C20 which itself was a bit hacky. Refactoring this in a way that you suggested is definitely cleaner approach, i am just not sure if we could solve the `is_alerts_query` in the same way. 
we can do that either in this or in a follow up PR. cc @jan-auer

Reply:
Sounds good to me!

Reply:
Sounds good to me!

Reply:
I remember now that I have added `is_alerts_query` after a suggestion by the owner of that code. At first I did implement like I suggested but I think it was turned down for "added complexity".